lib/galaxy/tool_util/xsd/galaxy.xsd

```diff
@@ -717,6 +718,47 @@ Read more about configuring Galaxy to run Docker jobs
       </xs:extension>
     </xs:simpleContent>
   </xs:complexType>
+  <xs:complexType name="Secret">
+    <xs:annotation>
+      <xs:documentation xml:lang="en"><![CDATA[
+This is possible to add secrets into tools directly by adding a requirement of
+type ``secret``. This will inject the value of the requirement into the
+environment of the tool. The value of the requirement is the value of the
+environment variable to inject the secret into.
+
+```xml
+<requirements>
+    <secret type="vault" user_preferences_key="some_tool|api_key" inject_as_env="some_tool_api_key" label="API Key" required="true"/>
```

Comment:
Do any of the use cases involve multiple tools with different IDs requesting the same secret? If not - I would make the tool id (without a version) an implicit prefix here and just attach some similar field as the suffix. This isolation feels more secure-ish to me - we wouldn't risk tools picking up other keys accidentally.

Reply:
Well, in some use cases, maybe in the future, it will be good to access an api_key or a token for multiple tools.

Reply:
Hi John, Arash, with @Marie59 and @jeremyfix we have such use case (if I well understand) where separate tools need the same credential (here this is Copernicus marine system credentials), so I confirm that this is already a reality ;) It appears to me, at least for ecology and related environmental sciences, that it will be the case for many data providers as Copernicus for satellites remote sensing data and others.